app/helpers/application_helper/button/basic.rb

```diff
@@ -12,6 +12,13 @@ def initialize(view_context, view_binding, instance_data, props)
     end
   end
 
+  def check_rbac?
+    return true if self[:child_id].nil?
+    # TODO: remove this when these buttons will be refactored
+    return true if %w(download_csv download_text custom_button).include?(self[:child_id])
```

Comment:
this looks wrong.

for one: it's something very specific to be in the base class

second:
```
devil - [~/Projects/manageiq] (master)$ grep -r download_csv app/helpers/application_helper
app/helpers/application_helper/toolbar/x_gtl_view.rb:          :download_csv,
app/helpers/application_helper/toolbar/gtl_view.rb:          :download_csv,
app/helpers/application_helper/toolbar/drift_view.rb:          :drift_download_csv,
app/helpers/application_helper/toolbar/compare_view.rb:          :compare_download_csv,
app/helpers/application_helper/toolbar/chargeback_center.rb:          :chargeback_download_csv,
app/helpers/application_helper/toolbar/miq_capacity_view.rb:          :miq_capacity_download_csv,
app/helpers/application_helper/toolbar_builder.rb:      if role_allows?(:feature => "chargeback_reports") && ["chargeback_download_csv", "chargeback_download_pdf",
app/helpers/application_helper/toolbar_builder.rb:      refresh_log fetch_log common_drift download_text download_csv download_pdf download_view vm_download_pdf
```

What about adding the missing features? Ping @dclarizio